lab-davide/test/note-router-test.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+process.env.MONGO_URI = 'mongodb://localhost/test';
+
+//npm module//
+const debug = require('debug')('note:note-tests');
+const expect = require('chai').expect;
+const request = require('superagent-use');
+const superPromise = require('superagent-promise-plugin');
+
+
+request.use(superPromise);
+
+
+//app modules//
+const server = require('../server');
+const noteCrud = require('../lib/note-crud');
+
+//required modules dependent on global env//
+const port = process.env.PORT || 3000;
+const baseUrl = `http://localhost:${port}/api/note`;
+
+
+//server is running//
+describe('testing module note-router', function() {
+  debug('hitting module note router');
+  before((done) => {
+    if (!server.isRunning) {
+      server.listen(port, () => {
+        debug('server running on port', port);
+        server.isRunning = true;
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if(server.isRunning) {
+      debug('server is close');
+      server.close(() => {
+        server.isRunning = false;
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing GET /api/note/:id with valid id', function(){
+    before((done) => {
+      noteCrud.createNote({name: 'test note', content: 'test data'})
+      .then(note => {
+        debug('Created note for GET tests');
+        this.tempNote = note;
+        done();
+      })
+      .catch(() => {
```

Comment:
If you got an error, I would definitely take it in as an argument so that you can log it and see what went wrong.